py/client/pydeephaven/_app_service.py

```diff
@@ -0,0 +1,20 @@
+#
+#  Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+from pydeephaven.dherror import DHError
+from pydeephaven.proto import application_pb2_grpc, application_pb2
+
+class AppService:
+    def __init__(self, session):
+        self.session = session
+        self._grpc_app_stub = application_pb2_grpc.ApplicationServiceStub(session.grpc_channel)
+
+    def list_fields(self):
+        try:
+            fields = self._grpc_app_stub.ListFields(
+                application_pb2.ListFieldsRequest(),
+                metadata=self.session.grpc_metadata
+            )
+            return fields
+        except Exception as e:
+            raise DHError("failed to list fields.") from e
```

Comment:
Not to distract from this PR - but I think we might want to look into the python grpc asyncio APIs - it might be a more natural fit for streaming responses like this. @jmao-denver